### PR TITLE
drivers: gpio: Migrate to new logging subsys

### PR DESCRIPTION
**Pull request by Olivier-ProGlove**
Migrate from `SYS_LOG` to `LOG` logging mechanism.

Signed-off-by: Olivier Martin <olivier.martin@proglove.de>

### CODE REVIEW

**Comment by Olivier-ProGlove** [comment]
I have rebased the pull-request to fix the conflict after this pull-request has been merged https://github.com/zephyrproject-rtos/zephyr/commit/8cf8db3a73aa0658d728006e93d9ef5e0361a1b2

File: drivers/gpio/gpio_pcal9535a.c [hunk 1]
```diff
@@ -351,7 +351,7 @@ static int gpio_pcal9535a_config(struct device *dev, int access_op,
 {
 	int ret;
 
-#if (CONFIG_SYS_LOG_GPIO_PCAL9535A_LEVEL >= SYS_LOG_LEVEL_DEBUG)
+#if (CONFIG_LOG_GPIO_LEVEL >= LOG_LEVEL_DEBUG)
```
Comment (nordic-krch) [inline drivers/gpio/gpio_pcal9535a.c]: you should be able to remove this #ifdef, compiler will cut it if logs are gone.

**Review by nordic-krch — CHANGES_REQUESTED** [review]
I am missing registration of log module (LOG_MODULE_REGISTER)

**Comment by nashif** [comment]
addressed in #10029

**Pull request closed without merging**